test/e2e/tests/test_queue.py

```diff
@@ -14,84 +14,114 @@
-                QueueName=queue_name
-            )
-            return resp
-
-        except Exception as e:
-            logging.debug(e)
-            return None
-
-    def queue_exists(self, sqs_client, queue_name: str) -> bool:
-        return self.get_queue(sqs_client, queue_name) is not None
-
-    def test_smoke(self, sqs_client):
-        resource_name = random_suffix_name("sqs-queue", 24)
-
-        resources = get_bootstrap_resources()
-        logging.debug(resources)
-
-        replacements = REPLACEMENT_VALUES.copy()
-        replacements["QUEUE_NAME"] = resource_name
-        replacements["AWS_REGION"] = get_region()
-
-        # Load Queue CR
-        resource_data = load_sqs_resource(
-            "queue",
-            additional_replacements=replacements,
-        )
-        logging.debug(resource_data)
+@pytest.fixture(scope="module")
+def simple_queue():
+    resource_name = random_suffix_name("sqs-queue", 24)
 
-        # Create k8s resource
-        ref = k8s.CustomResourceReference(
-            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
-            resource_name, namespace="default",
-        )
-        k8s.create_custom_resource(ref, resource_data)
-        cr = k8s.wait_resource_consumed_by_controller(ref)
+    resources = get_bootstrap_resources()
+    logging.debug(resources)
 
-        assert cr is not None
-        assert k8s.get_resource_exists(ref)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["QUEUE_NAME"] = resource_name
 
-        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+    # Load Queue CR
+    resource_data = load_sqs_resource(
+        "queue",
+        additional_replacements=replacements,
+    )
+    logging.debug(resource_data)
+
+    # Create k8s resource
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+        resource_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+    cr = k8s.wait_resource_consumed_by_controller(ref)
 
-        cr = k8s.wait_resource_consumed_by_controller(ref)
+    sqsqueue.wait_until_exists(resource_name)
 
-        # Check Queue exists
-        exists = self.queue_exists(sqs_client, resource_name)
-        assert exists
+    yield cr, ref
 
-        # Delete k8s resource
-        _, deleted = k8s.delete_custom_resource(ref)
-        assert deleted is True
+    # Delete k8s resource
+    _, deleted = k8s.delete_custom_resource(
+        ref,
+        period_length=DELETE_WAIT_AFTER_SECONDS,
+    )
+    assert deleted
 
-        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+    sqsqueue.wait_until_deleted(resource_name)
 
-        # Check Queue doesn't exist
-        exists = self.queue_exists(sqs_client, resource_name)
-        assert not exists
\ No newline at end of file
+
+@service_marker
+@pytest.mark.canary
+class TestQueue:
+    def test_crud(self, simple_queue):
+        res, ref = simple_queue
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
```

Comment:
Nit: what is the point of this, given that you've already done `sqsqueue.wait_until_exists(resource_name)` in the fixture setup above?

Reply:
I've noticed there can be a slight delay before the tags information is accurately returned for certain objects, including Queues...